staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -63,6 +63,10 @@ type Reflector struct {
 	store Store
 	// listerWatcher is used to perform lists and watches.
 	listerWatcher ListerWatcher
+	// forcePagination is used to always attempt paginating list requests.
+	// It may result in forcing the requests to be served from etcd instead of
+	// watch cache, so should be used carefully.
+	forcePagination bool
```

Comment:
the reflector already has an option for WatchListPageSize, which the watch cache specifically sets... should just honor that and avoid this new option? having both makes this pretty confusing

Reply:
> The reflector already has an option for WatchListPageSize, which the watch cache is the only client of... should just honor that and avoid this new option? having both makes this pretty confusing

I was afraid of it, because we want to treat consistent lists differently than "at least as fresh as" ones.
But after thinking more about that, it actually translates to the same thing.
Changing...